SalaryService.Application/Queries/CompensationsQuery.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SalaryService.Application.Queries.Contracts;
+using SalaryService.DataAccess;
+using SalaryService.Domain;
+
+namespace SalaryService.Application.Queries;
+
+public class CompensationsQuery : ICompensationsQuery
+{
+    private readonly EmployeeDbContext _context;
+
+    public CompensationsQuery(EmployeeDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<IEnumerable<Compensation>> GetCompensationsAsync()
+    {
+        return await _context
+            .Compensations
+            .ToListAsync();
```

Comment:
you need to add .AsNoTracking() because you aren't going to track these entities later